plugins/qiskit_executor/tasks.py

```diff
@@ -58,7 +57,73 @@ def start_execution(self, db_id: int) -> str:
         db_task.data["parameters"]
     )
 
-    TASK_LOGGER.info(f"Start execution with parameters: {str(circuit_params)}")
+    backend_list = None
+    if (
+        not circuit_params.ibmqToken
+        or (backend_list := get_backend_names(circuit_params.ibmqToken)) is None
+    ):
+        # start the authentication task
+        href = PluginState.get_value(
+            QiskitExecutor.instance.identifier, f"authentication_href_{db_task.id}"
+        )
+        ui_href = PluginState.get_value(
+            QiskitExecutor.instance.identifier, f"authentication_ui_href_{db_task.id}"
+        )
+        msg = "Started authentication task"
+        if circuit_params.ibmqToken:
+            msg += " (invalid IBMQ token)"
+        self.replace(
+            add_step.s(
+                task_log=msg,
+                db_id=db_task.id,
+                step_id="authentication",
+                href=href,
+                ui_href=ui_href,
+                prog_value=1,
+                prog_target=2,
+            )
+        )
+        return msg
+
+    PluginState.set_value(
+        QiskitExecutor.instance.identifier,
+        f"backend_list_{db_task.id}",
+        backend_list,
+        commit=True,
+    )
+
+    backend = None
+    if (
+        not circuit_params.backend
+        or (
+            backend := get_qiskit_backend(
+                circuit_params.backend, circuit_params.ibmqToken
+            )
+        )
+        is None
+    ):
+        # start the backend selection task
+        href = PluginState.get_value(
+            QiskitExecutor.instance.identifier, f"backend_selection_href_{db_task.id}"
+        )
+        ui_href = PluginState.get_value(
+            QiskitExecutor.instance.identifier, f"backend_selection_ui_href_{db_task.id}"
+        )
+        msg = "Started backend selection task"
+        if circuit_params.backend:
+            msg += " (invalid backend)"
+        self.replace(
+            add_step.s(
+                task_log=msg,
+                db_id=db_task.id,
+                step_id="backend-selection",
+                href=href,
+                ui_href=ui_href,
+                prog_value=1,
+                prog_target=2,
```

Comment:
Read previous progress and adapt numbers as required for better progress feedback for the user.

Reply:
The plugin has 3 possible steps:
1. circuit selection
2. authentication (if not done in step 1)
3. backend selection (if not done in step 1 or 2)

I updated the progress indication such that the `progress_value` shows the number of successfully executed steps (0-3) and the `progress_target` shows the minimum number of steps that have to be done (1-3).
When the user enters neither `ibmqToken` not `backend`, the progress indicator shows `1/2`, since we don't know if step 3 is required or if the user enters the `backend` in the authentication step.
Failed steps (invalid token/backend) do not increase `progress_value` or `progress_target`.

Reply:
This one is in the if for the missing backend so it should already be step 2 or 3 depending on wether the ibm token was provided in the first step or the second one. You could also always say that this is step 3 (even if step 2 was skipped), then progress for this step would just be 2/3.

Reply:
Now, the authentication step shows `1/3 steps` (only circuit selection is done) and the backend selection shows `2/3` steps done (circuit selection and authentication done).